scripts/check-one-feature

```diff
@@ -12,21 +12,36 @@ fi
 feature=$1
 shift
 
-if [ ! -d target ]
+# We assume that `target` is the normal target directory.
+base_target_dir="target"
+
+# Check to see if we have the nightly toolchain installed, and if so, do a deeper check to see if
+# there's a custom target directory configured.
+if cargo +nightly 2>&1 >/dev/null; then
+  overridden_base_target_dir=$(cargo +nightly -Zunstable-options config get | grep "build.target-dir" | sed 's/build.target-dir = //' | sed 's/\"//g' || true)
+  if [ ! -z "$overridden_base_target_dir" ]; then
+    base_target_dir=$overridden_base_target_dir
+  fi
+fi
+
+# Make sure that `cargo check` has been run so that we have something to bootstrap our per-feature
+# target directory from.
+if [ ! -d $base_target_dir/debug ]
 then
-  echo $0: 'Run `cargo check` first to prime the `target` directory.'
+  echo "$0: Run \`cargo check\` first to prime the \`${base_target_dir}\` directory."
   exit 1
 fi
 
-target=$PWD/target-feature-$feature
+target=$PWD/$base_target_dir/component-features/feature-$feature
+mkdir -p $target
 log=$target/.logfile
 trap 'rm --force --recursive $target' EXIT
 
-cp --archive --link target $target
-rm --force $target/debug/.cargo-lock
+cp --archive --link $base_target_dir/debug $target
+rm --force $base_target_dir/debug/.cargo-lock
 
 echo "===== Feature: $feature ====="
-if cargo check --workspace --all-targets \
```

Comment:
👍 to this, but it caused me to notice that I don't think we are actually checking workspace members' feature flags. I wonder if we want to? That can be a follow up.

Reply:
Right, the optimization we're doing to filter down to feature flags we care about are inherently oriented towards the `vector` crate, which means, at best, we were indirectly checking and discovering mismatches between, say... two crates importing the same dependency but with different feature flags. That's what was happening to Spencer and his PR to move the OTEL stuff to a crate under `lib/`, which prompted me to do this PR in the first place.

But other than those indirect discoveries by using `--workspace`, we're never feeding feature flags that are specific to those workspace crates, unless it's happening via the feature flags in `vector` itself.

Reply:
This is all a long way of saying that I think, even with this change, we're still testing what we want to test, even if we're now ignoring secondary compilation issues that are indeed real issues but only when viewed from a contrived scenario (like building a dependency that should not actually be included based on the provided feature flags).

Reply:
Yeah, I buy that argument since the subcrates aren't used outside of Vector itself.